src/app-layout/mobile-toolbar/index.tsx

```diff
@@ -130,33 +129,41 @@ export function MobileToolbar({
           aria-label={drawers.ariaLabel}
           className={clsx(styles['drawers-container'], testutilStyles['drawers-mobile-triggers-container'])}
         >
-          {visibleItems?.map((item: DrawerItem, index: number) => (
-            <span className={clsx(styles['mobile-toggle'], styles['mobile-toggle-type-drawer'])} key={index}>
+          {visibleItems.map((item: DrawerItem, index: number) => (
+            <div
+              className={clsx(styles['mobile-toggle'], styles['mobile-toggle-type-drawer'])}
+              key={index}
+              onClick={() => drawers.onChange({ activeDrawerId: item.id })}
+            >
               <ToggleButton
                 className={testutilStyles['drawers-trigger']}
                 iconName={item.trigger.iconName}
                 iconSvg={item.trigger.iconSvg}
                 badge={item.badge}
                 ariaLabel={item.ariaLabels?.triggerButton}
-                onClick={() => drawers.onChange({ activeDrawerId: item.id })}
                 ariaExpanded={drawers.activeDrawerId !== undefined}
                 testId={`awsui-app-layout-trigger-${item.id}`}
               />
-            </span>
+            </div>
```

Comment:
Changed from `<span>` to `<div>` in a few places, because button dropdown renders a `<div>` and it cannot be nested into `<span>`, so we need to upgrade all elements to div all way up